test/ngResource/resourceSpec.js

```diff
@@ -2101,6 +2101,28 @@ describe('cancelling requests', function() {
 
     expect(creditCard.$cancelRequest).toBe(noop);
   });
+
+  it('should not break when calling old `$cancelRequest` after the response arrives', function() {
+    $httpBackend.whenGET('/CreditCard').respond({});
+
+    var CreditCard = $resource('/CreditCard', {}, {
+      get: {
+        method: 'GET',
+        cancellable: true
+      }
+    });
+
+    var creditCard = CreditCard.get();
+    var cancelRequest = creditCard.$cancelRequest;
+
+    creditCard.$promise.then(function(list) {
+        cancelRequest();
```

Comment:
If you call `cancelRequest` after flushing (as suggested in https://github.com/angular/angular.js/pull/16037/files#r127979323), then this shouldn't be needed (unless I am missing something).